teacher.rb

```diff
@@ -0,0 +1,12 @@
+require './person'
+
+class Teacher < Person
+  def initialize(specialization, age, name = 'unknown', parent_permission: true)
+    @specialization = specialization
+    super(age, name, parent_permission: parent_permission)
+  end
```

Comment:
- [ Required ] Please consider making the call to the `superclass` constructor the first statement in your `base class`, this is usually the norm in most Object Oriented programming languages.

```RUBY
  def initialize(specialization, age, name = 'unknown', parent_permission: true)
       super(age, name, parent_permission: parent_permission)
      @specialization = specialization
  end
```

Reply:
I have also gone ahead to implement the change that you required here. The code looks more conventional now, all thanks to you. 💯